srml/fees/src/lib.rs

```diff
@@ -47,13 +72,17 @@ decl_module! {
 
 		fn on_finalise() {
```

Comment:
This could be done more cleanly by adding an event hook `OnTransactionFeePaid` into executive (or, probably better, `system::Trait` which would be called from executive) which could then be hooked in to `fees::Trait::OnFeeChanged`

Reply:
I believe that is basically the `ChargeFee` trait. Storage write (or some in memory write) is necessary to combine multiple fees together and emit a single event with the accumulated fee (#1815). I don't think there is other way to achieve that in the current substrate.

Reply:
In-memory write is way faster than storage round-trip, so I would prefer it to be donee thtat way, especially since it will affect all chains, even those with no use for this fee mechanism.

Reply:
Make sense. Is there existing in memory store can be used? Or need to crate this new  mechanism? which will be out of scope of this PR and shouldn’t prevent this PR from been merged.